### ISSUE TITLE
Failed certificate validation on API subdomain when using Notify API

### ISSUE THREAD

**Issue by jimleroyer** [cds-snc/notification-planning#624]
## Describe the bug 

A user from the API makes a request to a secured HTTPS connection to the Notify API and the security certificate fails validation, hence making usage of Notify API impossible in a secured way.

From a [reported issue](https://cds-snc.freshdesk.com/a/tickets/9116):

> Weblogic is rejecting the wildcarded notification.canada.ca certificate on the API hostname: <BEA-090504> <Certificate chain received from proxy.omega.dce-eir.net - --> api.notification.canada.ca failed hostname verification check. Certificate contained notification.canada.ca but check expected api.notification.canada.ca> Is it possible to explicitly add api.notification.canada.ca (non-wildcarded) as a subject alternative name on the certificate?

## [Bug Severity](https://docs.google.com/document/d/1tcD32bjJRaREsMgaUEoXAk400y1dKzzaKR0G5Sr1w20/edit)

SEV2, SEV3 -- need to discuss with team as impact might depend

## To Reproduce

1. Make a request to Notify API with certificate validation enabled. Potentially, it might happen only with some library that makes more exhaustive checks such as WebLogic as described by the reporting user [in Freshdesk](https://cds-snc.freshdesk.com/a/tickets/9116).

## Expected behavior

No certificate validation fails when communicating with Notify API.

## Impact

I am unsure what's the real impact, as only one user reported it so far.

Impact on Notify users: Not being able to use the Notify API without disabling the security certificate validation.

Impact on Recipients: Potentially notifications that were not sent all the way through with the best security measure in place. Less trust if discovered and exploited.

Impact on Notify team:  Potentially less trust when something bad happens.

## Additional context

From @mohamed-cds 

> this being [straightforward as updating the acm terraform](https://github.com/cds-snc/notification-terraform/blob/0be5c3eaa802af68c46b109d4102c33d1935b8c8/aws/dns/acm.tf#L3-L6) to add a new alt name.

```
subject_alternative_names = [
    "*.${var.domain}",
    "*.document.${var.domain}"
  ]
```

**Comment by mohamed-cds** [comment]
just a heads up that `subject_alternative_names` appears twice in the code, so both locations would have to be updated. An even better solution would be to find a way to use a common local variable so that it doesn't need to be updated in multiple places.

**Comment by jimleroyer** [comment]
Client [responded back on the ticket](https://cds-snc.freshdesk.com/a/tickets/9116), saying they might need this sooner than later. [Opened a PR](https://app.zenhub.com/workspaces/notify-planning-614b3ad91bc2030015ed22f5/issues/cds-snc/notification-planning/624) for that to test it out.

**Comment by jimleroyer** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Notify-Planning-614b3ad91bc2030015ed22f5/issues/cds-snc/notification-planning/624?planning-poker) with ZenHub @mohamed-cds @sastels

**Comment by yaelberger-commits** [comment]
If the size of this task is relatively small, we could bring this in for the next sprint to tackle within the next 3 weeks to meet the client's needs.

**Comment by jimleroyer** [comment]
We tested the change in staging and it didn't work as expected (and estimated -- this will be more points). This requires the SRE team intervention. @mohamed-cds kindly volunteered to help us on that one, bringing a solution on the table that will unblock us on this. There is no hurry as we'd have 1 month but they might have time for us this week to push forward.

**Comment by mohamed-cds** [comment]
What i propose is we split this into two tasks:
1. Getting the current cert update into production
2. Refactoring terraform so that alt names can be added without requiring manual intervention

What we did for Item `#1`:
- Merge https://github.com/cds-snc/notification-terraform/pull/493. The terraform apply will create the new certificate but cannot delete the old since it's still in use by the Application Load balancer and the API Gateway
- Create CNAME validation records for the new cert (SRE Tools route53 for staging, and DNS repo for production)
- Using the AWS console update the:
   - Application load balancer to use the new certificate after it's been successfully updated
   - API Gateway custom domains to use the new certificate
- Wait for certificate updates to be completed (API Gateway may take 10-15 minutes)
- Rerun failed terraform apply

**Comment by jimleroyer** [comment]
This is planned to go in production on July 21th, 2022, 16h30 EST.

**Comment by adriannelee** [comment]
Jimmy:

> I am currently working on the WAF rules, the card is in progress, no blocker at the moment! Also, the task to upgrade the certificate might be blocked at the moment, the user came back to say it does not work. I think the software he is using (i.e. WebLogic) might need some tweak so I'll contact him again on Freshdesk.

**Comment by adriannelee** [comment]
Blocked by client and their configuration. We'll need to negotiate how much we change our configuration vs them changing theirs

**Comment by jimleroyer** [comment]
Waiting for users to reply back as per previous Adrianne's comment. We'll see what they say on the suggestion we have for them with a tweak to their configuration and then concert on options.

**Comment by jimleroyer** [comment]
User replied back and proposed to add a new entry in addition to the wildcard one, so basically wildcard + non-wildcard. Spoke to Max and he said let's try this on.
https://github.com/cds-snc/notification-terraform/pull/500

**Comment by jimleroyer** [comment]
We pushed new certificates in staging and will make a push in production on August 2 in the morning. CC ops lead @sastels

**Comment by yaelberger-commits** [comment]
Release tomorrow morning

**Comment by sastels** [comment]
LGTM. Suggest marking this as Done until / unless we hear back from the user that they're still having issues.
